flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/fs/ForStFlinkFileSystem.java

```diff
@@ -101,6 +156,13 @@ public boolean rename(Path src, Path dst) throws IOException {
         // The rename is not atomic for ForSt. Some FileSystems e.g. HDFS, OSS does not allow a
         // renaming if the target already exists. So, we delete the target before attempting the
         // rename.
+
+        if (localFileFilter.apply(src.getName())) {
+            Path localSrc = tryBuildLocalPath(src).f1;
+            Path localDst = tryBuildLocalPath(dst).f1;
+            return localFS.rename(localSrc, localDst);
```

Comment:
if this is directory, the remote path should also be renamed.  I'd suggest check the filestatus first

Reply:
Good point, fixed.